lib/routes/account.js

```diff
@@ -591,7 +607,8 @@ module.exports = function (
           //  * the request wants keys, since unverified sessions are fine to use for e.g. oauth login.
           //  * the email is verified, since content-server triggers a resend of the verification
           //    email on unverified accounts, which doubles as sign-in confirmation.
-          if (didSigninUnblock || !features.isSigninConfirmationEnabledForUser(emailRecord.uid, emailRecord.email, request)) {
+          //  * the login is coming from an ip address not previously seen.
+          if (didSigninUnblock || !features.isSigninConfirmationEnabledForUser(emailRecord.uid, emailRecord.email, request) || hasVerifiedSessionFromIp) {
```

Comment:
I can get behind that, geo-profiling and ua-agent checks could eventually live in there. Although maybe that might be better in a separate module.